flink-connector-jdbc-core/src/main/java/org/apache/flink/connector/jdbc/core/util/Precondition.java

```diff
@@ -0,0 +1,290 @@
+package org.apache.flink.connector.jdbc.core.util;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import javax.annotation.Nullable;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * A collection of static utility methods to validate input.
+ *
+ * <p>This class is a copy of Flink Preconditions. We add this code to the Flink JDBC code because
+ * Preconditions is an Internal class of Flink.
+ */
+@PublicEvolving
+public class Precondition {
```

Comment:
If we ultimately keep this class, what problems might arise from using the class name Preconditions?
What I mean is that since the comment mentions this is a code copy, perhaps we can just use the package name to differentiate it and keep the original class-name.

Reply:
None, I simple give another name to be more easy to find the right class